server/src/main/java/org/elasticsearch/action/support/IndicesOptions.java

```diff
@@ -103,7 +102,8 @@ public static XContentBuilder toXContent(EnumSet<WildcardStates> states, XConten
         ALLOW_NO_INDICES,
         FORBID_ALIASES_TO_MULTIPLE_INDICES,
         FORBID_CLOSED_INDICES,
-        IGNORE_THROTTLED;
+        IGNORE_THROTTLED,
+        INCLUDE_DATA_STREAMS;
```

Comment:
> It looks like only search, refresh (and index) really need to resolve data streams?

I think more apis will need to be able to resolve data streams. I just enabled the ability for the APIs you mentioned in order to make integration tests pass. For example I think that field capability, get field mapping and eql APIs will need to resolve data streams as well. 

Also I suspect that get mapping and settings apis will need to be able to resolve data streams too (if a data stream is used then all mappings/settings for all backing indices are returned.) When talking to the Fleet team, they would like to the put mapping/settings apis to understand data streams as well (in this case a data stream should resolve to the latest index, otherwise they will need to figure out what the current write index is and do the required update). This was just part of casual conversation, so no decisions made yet.

I'm pretty sure that more apis will need to support data streams, so 
I prefer if we keep the current approach that enables the use data streams via an `Indices.Option` enum instance. Also the way I look at data streams is that it is a way to group indices and from that perspective exposing the ability to use data streams in apis as an indices option makes sense. I don't think we should ever hide the fact that a data stream is backed by indices. 

Only part of indices option is user specified. The include data streams indices option is not a user specified option as apposed to example allow_no_indices option. Each api/action can opt-in whether data streams should be resolvable. I think adding a method to `IndexNameExpressionResolver` will complicate the code. Now most apis use the same method for resolving user specified indices expressions and for all apis that need to be able support data streams this needs to be changed. Also apis tend to reuse the same base class.

Reply:
We spoke about this on another channel and came to the conclusion to keep this as is. There is agreement that a split of the user level and the action-specific options would be desirable, but we postpone that work to a separate effort.